test/middleware/index.js

```diff
@@ -0,0 +1,64 @@
+const { expect } = require("chai");
+const sinon = require("sinon");
+const { directMessage, anyOf, reactionMatches } = require("../../middleware");
+
+describe("middleware/index", () => {
+  describe("directMessage", () => {
+    it('should call next if message channel type is "im"', async () => {
+      const next = sinon.spy();
+      const message = { channel_type: "im" };
+      await directMessage()({ message, next });
+      expect(next.calledOnce).to.be.true;
+    });
+
+    it('should not call next if message channel type is not "im"', async () => {
+      const next = sinon.spy();
+      const message = { channel_type: "not_im" };
+      await directMessage()({ message, next });
+      expect(next.called).to.be.false;
+    });
+  });
+
+  describe("anyOf", () => {
+    it("should call next if any of the passed functions return true", async () => {
```

Comment:
Based on this description, I'd expect either `func1` or `func2` to return true, but they're both returning `undefined`.